internal/locate/region_cache.go

```diff
@@ -2468,8 +2468,9 @@ func (r *Region) Contains(key []byte) bool {
 // for the maximum region endKey is empty.
 // startKey < key <= endKey.
 func (r *Region) ContainsByEnd(key []byte) bool {
-	return bytes.Compare(r.meta.GetStartKey(), key) < 0 &&
-		(bytes.Compare(key, r.meta.GetEndKey()) <= 0 || len(r.meta.GetEndKey()) == 0)
+	return len(key) == 0 && len(r.EndKey()) == 0 ||
+		bytes.Compare(r.meta.GetStartKey(), key) < 0 &&
+			(bytes.Compare(key, r.meta.GetEndKey()) <= 0 || len(r.meta.GetEndKey()) == 0)
```

Comment:
maybe this will make it easier to read:
```
if len(key) == 0 {
    return len(r.EndKey()) == 0
}
return bytes.Compare(r.meta.GetStartKey(), key) < 0 &&
			(bytes.Compare(key, r.meta.GetEndKey()) <= 0 || len(r.meta.GetEndKey()) == 0)
```